src/core/public/chat/chat_service.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import { CoreService } from '../../types';
+import {
+  ChatServiceSetup,
+  ChatServiceStart,
+  ChatImplementationFunctions,
+  Message,
+  ChatWindowState,
+} from './types';
+
+/**
+ * Core chat service - manages infrastructure state
+ */
+export class ChatService implements CoreService<ChatServiceSetup, ChatServiceStart> {
+  private implementation?: ChatImplementationFunctions;
+  private suggestedActionsService?: { registerProvider(provider: any): void };
+
+  // Core-managed infrastructure state
+  private threadId$ = new BehaviorSubject<string>(this.generateThreadId());
+  private windowState$ = new BehaviorSubject<ChatWindowState>({
+    isWindowOpen: false,
+    windowMode: 'sidecar',
+    paddingSize: 400,
+  });
+  private windowOpenCallbacks = new Set<() => void>();
+  private windowCloseCallbacks = new Set<() => void>();
+
+  private generateThreadId(): string {
+    const timestamp = Date.now();
+    const randomStr = Math.random().toString(36).substring(2, 11);
+    return `thread-${timestamp}-${randomStr}`;
+  }
+
+  public setup(): ChatServiceSetup {
+    return {
+      setImplementation: (implementation: ChatImplementationFunctions) => {
+        this.implementation = implementation;
+      },
+
+      setSuggestedActionsService: (service: { registerProvider(provider: any): void }) => {
+        this.suggestedActionsService = service;
+      },
+
+      suggestedActionsService: this.suggestedActionsService,
+    };
+  }
+
+  public start(): ChatServiceStart {
+    const setWindowState = (partialState: Partial<ChatWindowState>) => {
+      const currentState = this.windowState$.getValue();
+      const newState = { ...currentState, ...partialState };
+      this.windowState$.next(newState);
+    };
+
+    const chatServiceInstance = this;
+
+    return {
+      // Availability check
+      isAvailable: () => {
+        return !!this.implementation;
+      },
+
+      // Thread management (core-managed)
+      getThreadId: () => {
+        return this.threadId$.getValue();
+      },
+
+      getThreadId$: () => {
+        return this.threadId$.asObservable();
+      },
+
+      setThreadId: (threadId: string) => {
+        this.threadId$.next(threadId);
+      },
+
+      newThread: () => {
+        const newThreadId = this.generateThreadId();
+        this.threadId$.next(newThreadId);
+      },
+
+      // Window state management (core-managed)
+      isWindowOpen: () => {
+        return this.windowState$.getValue().isWindowOpen;
+      },
+
+      getWindowState: () => {
+        return this.windowState$.getValue();
+      },
+
+      getWindowState$: () => {
+        return this.windowState$.asObservable();
+      },
+
+      setWindowState,
```

Comment:
Shall we hide this interface? I am thinking exposing this method may give too much details of the module.

Reply:
Nice comment. Do you think I can work this as a follow up? windowState is mainly UI related like paddingSize, open and mode. The current pattern is like direct expose, similar to other plugins.

```
// Core exposes setter
  core.workspaces.setClient(client)

// Plugin uses it
  this.coreStart.workspaces.setClient(myClient);
```

We can unify the pattern later in the follow-up?